crypto/pki/pki.go

```diff
@@ -272,6 +291,35 @@ func (p *PKI) GenerateIntermediateCertificate(name string, rootCrt *x509.Certifi
 	return err
 }
 
+// GenerateSSHSigningKeys generates and encrypts a private key used for signing
+// SSH user certificates and a private key used for signing host certificates.
+func (p *PKI) GenerateSSHSigningKeys(password []byte) error {
+	var pubNames = []string{p.sshHostCert, p.sshUserCert}
+	var privNames = []string{p.sshHostKey, p.sshUserKey}
+	for i := 0; i < 2; i++ {
+		pub, priv, err := keys.GenerateDefaultKeyPair()
+		if err != nil {
+			return err
+		}
+		if _, ok := priv.(crypto.Signer); !ok {
+			return errors.Errorf("key of type %T is not a crypto.Signer", priv)
+		}
+		sshKey, err := ssh.NewPublicKey(pub)
+		if err != nil {
+			return errors.Wrapf(err, "error converting public key")
+		}
+		_, err = pemutil.Serialize(priv, pemutil.WithFilename(privNames[i]), pemutil.WithPassword(password))
+		if err != nil {
+			return err
+		}
+		if err = utils.WriteFile(pubNames[i], ssh.MarshalAuthorizedKey(sshKey), 0600); err != nil {
```

Comment:
a bit confusing because here you're writing a public key but the variable name you're using is `...cert`. Let's just be consistent, unless there is a good reason not to be.